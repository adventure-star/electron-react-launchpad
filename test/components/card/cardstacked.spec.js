import React from 'react';
import ReactTestRenderer from 'react-test-renderer';

import CardStacked from '../../../src/renderer/components/card/cardstacked.jsx'

function setupStandard() {
    const renderer = ReactTestRenderer.create(<CardStacked>Test content</CardStacked>);
    const component = renderer.toJSON();

    return {
        component
    };
}

describe('Card - Card stacked component', () => {
    it('should be of type div', () => {
        const { component } = setupStandard();
        expect(component.type).toBe('div');
    });

    it('should have only the class "card-stacked"', () => {
        const { component } = setupStandard();
        expect(component.props.className).toBe('card-stacked');
    });

    it('children should be the text "Test content"', () => {
        const { component } = setupStandard();
        expect(component.children[0]).toBe('Test content');
    });

    it('Matches/Renders a snapshot', () => {
        const { component } = setupStandard();
        expect(component).toMatchSnapshot();
    });
});
