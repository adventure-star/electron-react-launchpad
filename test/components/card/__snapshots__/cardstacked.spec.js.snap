// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card - Card stacked component Matches/Renders a snapshot 1`] = `
<div
  className="card-stacked"
>
  Test content
</div>
`;
